src/gasPriceOracle/oracle.ts

```diff
@@ -0,0 +1,98 @@
+import { BigNumber, providers } from "ethers";
+import { toBN } from "../utils";
+
+export type GasPriceEstimate = {
+  maxFeePerGas: BigNumber;
+  maxPriorityFeePerGas?: BigNumber;
+};
+
+type ValidatedFeeData = {
+  [idx: string]: BigNumber;
+  gasPrice: BigNumber;
+  maxFeePerGas: BigNumber;
+  maxPriorityFeePerGas: BigNumber;
+};
+
+interface GasPriceFeed {
+  (provider: providers.Provider): Promise<GasPriceEstimate>;
+}
+
+async function getProviderFeeData(provider: providers.Provider): Promise<ValidatedFeeData> {
+  // Start with safe defaults.
+  const feeData: ValidatedFeeData = {
+    gasPrice: toBN(Number.MAX_SAFE_INTEGER),
+    maxFeePerGas: toBN(Number.MAX_SAFE_INTEGER),
+    maxPriorityFeePerGas: toBN(Number.MAX_SAFE_INTEGER),
+    // lastBaseFeePerGas: toBN(Number.MAX_SAFE_INTEGER), @todo: ethers 5.7.x
+  };
+
+  try {
+    const response: providers.FeeData = await provider.getFeeData();
+    const expectedKeys: string[] = Object.keys(feeData);
+    Object.entries(response).forEach(([key, value]) => {
+      if (expectedKeys.includes(key) && BigNumber.isBigNumber(value) && value.gt(0)) {
+        feeData[key] = toBN(value);
+      }
+    });
+  } catch {
+    // No logging, so nothing to do here...
+  }
+
+  return feeData;
+}
+
+async function legacy(provider: providers.Provider): Promise<GasPriceEstimate> {
+  const feeData: ValidatedFeeData = await getProviderFeeData(provider);
+
+  return {
+    maxFeePerGas: feeData.gasPrice,
+  } as GasPriceEstimate;
+}
+
+// @todo: Update to ethers 5.7.x to access FeeData.lastBaseFeePerGas
```

Comment:
Why not go ahead and do this so we don't have to build a workaround?

Reply:
We seem to have a little bit of work to do there (https://github.com/across-protocol/sdk-v2/actions/runs/3135735485/jobs/5091770273#step:4:278), so I thought it best to decouple them.

The impact of proceeding as-is is that we will probably have a _minor_ over-estimation in total gas cost on some occasions - but it's within what we would plan to add with gas padding anyway. The necessary changes after upgrading to ethers 5.7.x are also pretty minor - should only be a couple of lines to modify in this file.